packages/cms-cli/commands/functions/test.js

```diff
@@ -0,0 +1,208 @@
+const express = require('express');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const { spawn } = require('child_process');
+const os = require('os');
+const {
+  addAccountOptions,
+  addConfigOptions,
+  setLogLevel,
+  getAccountId,
+  addUseEnvironmentOptions,
+} = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { logDebugInfo } = require('../../lib/debugInfo');
+const {
+  loadConfig,
+  validateConfig,
+  checkAndWarnGitInclusion,
+} = require('@hubspot/cms-lib');
+const { logger } = require('@hubspot/cms-lib/logger');
+const { handleExit } = require('@hubspot/cms-lib/lib/process');
+const { validateAccount } = require('../../lib/validation');
+
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath, options);
+  checkAndWarnGitInclusion();
+
+  if (!(validateConfig() && (await validateAccount(options)))) {
+    process.exit(1);
+  }
+};
+
+/*
+  ==== TODO ====
+  - Handle functions with no package.json
+  - Handle functions with multiple methods
+*/
+
+const installDeps = folderPath => {
+  const npmCmd = os.platform().startsWith('win') ? 'npm.cmd' : 'npm';
+
+  return new Promise((resolve, reject) => {
+    try {
+      const npmInstallProcess = spawn(npmCmd, ['i'], {
+        env: process.env,
+        cwd: folderPath,
+        stdio: 'inherit',
+      });
+
+      npmInstallProcess.on('exit', resolve);
+    } catch (e) {
+      reject(e);
+    }
+  });
+};
+
+const cleanupArtifacts = folderPath => {
+  fs.rmdirSync(`${folderPath}/node_modules`, { recursive: true });
+  fs.unlinkSync(`${folderPath}/package-lock.json`);
+};
+
+const loadEnvVars = folderPath => {
+  const dotEnvPathMaybe = `${folderPath}/.env`;
+
+  if (fs.existsSync(dotEnvPathMaybe)) {
+    return require('dotenv').config({ path: dotEnvPathMaybe });
+  }
+
+  return {};
+};
+
+const runTestServer = async (port, accountId, functionPath) => {
+  const { endpoints, environment } = JSON.parse(
+    fs.readFileSync(`${functionPath}/serverless.json`, {
+      encoding: 'utf-8',
+    })
+  );
+  const routes = Object.keys(endpoints);
+
+  if (!routes.length) {
+    logger.error(`No endpoints found in ${functionPath}/serverless.json.`);
+    return;
+  }
+
+  await installDeps(functionPath);
+  handleExit(() => {
+    cleanupArtifacts(functionPath);
+  });
+
+  const app = express();
+  routes.forEach(route => {
+    const { method, file } = endpoints[route];
```

Comment:
we can also have environment variables configured for a specific route (which could also override the top-level environment variables)

Reply:
Local environment config will now be loaded and will overwrite global environment config.